plots.py

```diff
@@ -9,21 +10,25 @@
 )
 
 
-def plot_tips(sampled_data, log):
+def plot_tips(sampled_data, log, color="black"):
     plot = (
         ggplot(sampled_data, aes("tip_amount", "total_amount"))
-        + geom_point()
+        + geom_point(color=color)
         + theme_bw()
     )
     if log:
         plot = plot + scale_x_log10() + scale_y_log10()
-    return plot.draw()
+    fig = plot.draw()
+    close()
+    return fig
 
 
-def plot_hist(sampled_data):
+def plot_hist(sampled_data, color="black"):
     plot = (
         ggplot(sampled_data, aes(x="total_amount"))
-        + geom_histogram(binwidth=5)
+        + geom_histogram(binwidth=5, color=color, fill=color)
         + theme_bw()
     )
-    return plot.draw()
+    fig = plot.draw()
+    close()
```

Comment:
You will need to close the plot to not run into memory issues. After 20 plots generated Matplotlib will start raising warnings.